### PR TITLE
Add flash attention to speedup fused_gate_attention.

### PR DESCRIPTION
**Pull request by JamesLim-sy**
### PR types

Performance optimization
### PR changes

OPs
### Description

Pcard-70461

- The last optimization for AlphaFold models, which increasing the performance of AlphaFold2 model from 2.99s/iter to 2.796s/iter.

- Rely on : https://github.com/PaddlePaddle/flash-attention/pull/4

### CODE REVIEW

**Review by wanghuancoder — DISMISSED** [review]
LGTM

**Review by Shixiaowei02 — DISMISSED** [review]

**Review by luotao1 — DISMISSED** [review]

**Review by jiweibo — DISMISSED** [review]

**Review by Xreki — DISMISSED** [review]
LGTM

**Comment by Xreki** [comment]
try

**Comment by Xreki** [comment]
try

File: cmake/external/flashattn.cmake [hunk 1]
```diff
@@ -19,8 +19,8 @@ add_definitions(-DPADDLE_WITH_FLASHATTN)
 set(FLASHATTN_PREFIX_DIR ${THIRD_PARTY_PATH}/flashattn)
 set(FLASHATTN_SOURCE_SUBDIR csrc/flash_attn)
 set(FLASHATTN_INSTALL_DIR ${THIRD_PARTY_PATH}/install/flashattn)
-set(FLASHATTN_REPOSITORY ${GIT_URL}/PaddlePaddle/flash-attention.git)
-set(FLASHATTN_TAG 5ff4bbf56ad066750407c4aef16ac740ebda0717)
+set(FLASHATTN_REPOSITORY ${GIT_URL}/Xreki/flash-attention.git)
```
Comment (kuizhiqing) [inline cmake/external/flashattn.cmake]: Why use personal repo ?

cc @sneaxiy

Comment (Xreki) [reply cmake/external/flashattn.cmake]: 目前因包体积增大的比较多，在调试中，这是中间状态。

**Review by Xreki — APPROVED** [review]

**Review by jiweibo — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**